config/routes.rb

```diff
@@ -3,4 +3,7 @@
   root "welcome#index"
   get "/exchange_token", to: 'sessions#create'
   get "/dashboard", to: "dashboard#index"
+  patch "/dashboard", to: "dashboard#update"
+  get "/user/edit", to: "user#edit"
+  patch "/user/edit", to: "user#edit"
```

Comment:
patch '/user/edit/' should redirect to a user#update action that updates the User's username to the username entered